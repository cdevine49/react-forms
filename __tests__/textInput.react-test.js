import React from 'react';
import TextInput from '../src/components/textInput';
import renderer from 'react-test-renderer';

describe('TextInput', () => {
  test('default', () => {
    const tree = renderer.create(
      <TextInput />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('containerProps', () => {
    const tree = renderer.create(
      <TextInput
        containerProps={{
          id: 'container',
          className: 'classy react',
          style: { margin: "10px" }
        }}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('label', () => {
    const tree = renderer.create(
      <TextInput
        label="Child/Text"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('id', () => {
    const tree = renderer.create(
      <TextInput
        id="first-name"
        label="First Name"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('labelProps', () => {
    const tree = renderer.create(
      <TextInput
        label="Last Name"
        labelProps={{
          id: 'my-label',
          className: 'right large name',
          style: { fontSize: "16px" }
        }}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('className', () => {
    const tree = renderer.create(
      <TextInput
        className="big and fat"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('required', () => {
    const tree = renderer.create(
      <TextInput
        label="Gets an Asterisk"
        required={true}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('errorMessage', () => {
    const tree = renderer.create(
      <TextInput
        className="not overridden"
        errorMessage="Ya dun goofed"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('errorProps', () => {
    const tree = renderer.create(
      <TextInput
        errorMessage="Uh oh"
        errorProps={{
          id: 'error-tag',
          className: 'more classes',
          'data-message': 'hi there'
        }}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('underline', () => {
    const tree = renderer.create(
      <TextInput
        underline="More explanation"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('underlineProps', () => {
    const tree = renderer.create(
      <TextInput
        underline="Fill this in because ..."
        underlineProps={{
          id: "underline-tag",
          className: 'help explain',
          style: { background: 'blue' }
        }}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('other props', () => {
    const tree = renderer.create(
      <TextInput
        onFocus={jest.fn()}
        style={{ padding: "25px 15px" }}
        data-key="input-key"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('aria-describedby', () => {
    test('accepts both error id and underline id', () => {
      const tree = renderer.create(
        <TextInput
          errorProps={{
            id: 'error-id'
          }}
          underlineProps={{
            id: 'underline-id'
          }}
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});