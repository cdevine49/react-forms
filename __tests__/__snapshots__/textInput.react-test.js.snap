// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput ContainerProps className added to container tag's default classes 1`] = `
<div
  className="input-container classy react"
>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
</div>
`;

exports[`TextInput ContainerProps populate the container tag 1`] = `
<div
  className="input-container"
  id="container"
  style={
    Object {
      "margin": "10px",
    }
  }
>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
</div>
`;

exports[`TextInput Error errorMessage acts as child prop 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className="error"
    id={undefined}
  />
  <p
    className="error-message"
    role="alert"
  >
    Uh oh
  </p>
</div>
`;

exports[`TextInput Error errorProps populate the error tag 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className="error"
    id={undefined}
  />
  <p
    className="error-message more classes"
    data-message="hi there"
    id="error-tag"
    role="alert"
  >
    Uh oh
  </p>
</div>
`;

exports[`TextInput Input aria-describedby defaults to underlineProps.id 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby="descriptor"
    className={undefined}
    id={undefined}
  />
</div>
`;

exports[`TextInput Input aria-describedby will accept ariaDescribedby prop as a fallback 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby="secondary"
    className={undefined}
    id={undefined}
  />
</div>
`;

exports[`TextInput Input receives 'error' class when errorMessage prop is truthy 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className="error"
    id={undefined}
  />
  <p
    className="error-message"
    role="alert"
  />
</div>
`;

exports[`TextInput Input receives 'error' class when errorMessage prop is truthy 2`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className="big and fat error"
    id={undefined}
  />
  <p
    className="error-message"
    role="alert"
  >
    Ya dun goofed
  </p>
</div>
`;

exports[`TextInput Input receives 'error' class when errorMessage prop is truthy 3`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
</div>
`;

exports[`TextInput Input receives props by default 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className="big and fat"
    data-key="input-key"
    id="my-input"
    style={
      Object {
        "padding": "25px 15px",
      }
    }
  />
</div>
`;

exports[`TextInput Label htmlFor populated by id prop 1`] = `
<div
  className="input-container"
>
  <label
    htmlFor="first-name"
  >
    First Name
  </label>
  <input
    aria-describedby={undefined}
    className={undefined}
    id="first-name"
  />
</div>
`;

exports[`TextInput Label label prop acts as child 1`] = `
<div
  className="input-container"
>
  <label
    htmlFor={undefined}
  >
    Text Input
  </label>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
</div>
`;

exports[`TextInput Label labelProps populate the label tag 1`] = `
<div
  className="input-container"
>
  <label
    className="last name label"
    htmlFor={undefined}
    id="my-label"
    style={
      Object {
        "fontSize": "16px",
      }
    }
  >
    Last Name
  </label>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
</div>
`;

exports[`TextInput Underline id defaults to underlineProps.id 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby="descriptor"
    className={undefined}
    id={undefined}
  />
  <p
    className="underline"
    id="descriptor"
  >
    More info
  </p>
</div>
`;

exports[`TextInput Underline id will accept ariaDescribedby prop as a fallback 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby="secondary"
    className={undefined}
    id={undefined}
  />
  <p
    className="underline"
    id="secondary"
  >
    Extra info
  </p>
</div>
`;

exports[`TextInput Underline underline acts as child prop 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
  <p
    className="underline"
    id={undefined}
  >
    More explanation
  </p>
</div>
`;

exports[`TextInput Underline underlineProps populate the underline tag 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
  <p
    className="underline help explain"
    id={undefined}
    style={
      Object {
        "background": "blue",
      }
    }
  >
    Fill this in because ...
  </p>
</div>
`;

exports[`TextInput default 1`] = `
<div
  className="input-container"
>
  <input
    aria-describedby={undefined}
    className={undefined}
    id={undefined}
  />
</div>
`;
